src/NServiceBus.AmazonSQS/PreparedMessage.cs

```diff
@@ -0,0 +1,18 @@
+namespace NServiceBus.Transports.SQS
+{
+    using System.Collections.Generic;
+    using Amazon.SQS.Model;
+
+    class PreparedMessage
+    {
+        public string MessageId { get; set; }
+        public string Body{ get; set; }
+        public string Destination { get; set; }
+        public string OriginalDestination { get; set; }
+        public string QueueUrl { get; set; }
+        public long DelaySeconds { get; set; }
```

Comment:
For my curiosity what is the point of having `DelaySeconds` being a `long` if when used it need to be converted to an `int` anyway?

Reply:
I think I left it a long because it was a long before. Lemme check

Reply:
So for the calculation, we need it to be a long because we support with UnrestrictedDelayDelivery arbitrary long intervals. But given that prepared message is used to turn it into DelaySeconds of the type this can be an int and the conversation can happen when creating the prepared message